universal-application-tool-0.0.1/app/services/program/predicate/PredicateValue.java

```diff
@@ -46,16 +45,21 @@ public static PredicateValue of(ImmutableList<String> value) {
             .map(PredicateValue::surroundWithQuotes)
             .collect(toImmutableList())
             .toString(),
-        ScalarType.LIST_OF_STRINGS);
+        OperatorRightHandType.LIST_OF_STRINGS);
+  }
+
+  public static PredicateValue ofListOfLongs(ImmutableList<Long> value) {
+    return create(
+        value.stream().collect(toImmutableList()).toString(), OperatorRightHandType.LIST_OF_LONGS);
   }
 
   @JsonCreator
   private static PredicateValue create(
-      @JsonProperty("value") String value, @JsonProperty("type") ScalarType type) {
+      @JsonProperty("value") String value, @JsonProperty("type") OperatorRightHandType type) {
```

Comment:
question: is this backwards compatible? I think it _might_ be, since the enum values are the same and could be parsed. If not, we'll want a custom `@JsonCreator` to handle the legacy types

Reply:
Ohh great question. I was slightly worried about this as well. Let me see if I can figure out its backward compatibility.

Reply:
OK, after looking in the database and seeing that type is stored like `"type": "LIST_OF_STRINGS"`, etc, I am fairly confident this is backwards compatible for the reason you mentioned. All the existing `type`s should be able to be parsed as `OperatorRightHandType`s since that enum includes all the same names as `ScalarType` but just adds one more.